### PR TITLE
Creating powershell cmdlet for live debugging

### PR DESCRIPTION
**Pull request by LepiKlark**
First checkin in live debugging via powershell effort.

Contains project with CmdLet to start debug session and fetch callstacks.

Future work will take care of debugger control (stop, break, resume), breakpoints etc.

### CODE REVIEW

File: CsScriptManaged/CsScripts/Thread.cs [hunk 1]
```diff
@@ -133,7 +133,7 @@ public ThreadContext ThreadContext
         /// <summary>
         /// Gets the stack trace.
         /// </summary>
-        private StackTrace GetStackTrace()
+        public StackTrace GetStackTrace()
```
Comment (southpolenator) [inline CsScriptManaged/CsScripts/Thread.cs]: Don't make this public function. Use caching property named StackTrace.

File: CsScriptManaged/CsScripts/Thread.cs [hunk 2]
```diff
@@ -153,7 +153,7 @@ public static StackTrace GetStackTraceFromContext(ulong contextAddress, uint con
         /// <summary>
         /// Gets the thread context.
         /// </summary>
-        private ThreadContext GetThreadContext()
+        public ThreadContext GetThreadContext()
```
Comment (southpolenator) [inline CsScriptManaged/CsScripts/Thread.cs]: Same as for StackTrace

File: CsScriptManaged/CsScripts/Thread.cs (hunk 1 above)
Comment (LepiKlark) [inline CsScriptManaged/CsScripts/Thread.cs]: Done.

File: CsScriptManaged/CsScripts/Thread.cs (hunk 2 above)
Comment (LepiKlark) [inline CsScriptManaged/CsScripts/Thread.cs]: Done.

File: DebugCmdlets/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+//
+//      Major Version
+//      Minor Version 
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers 
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("1.0.0.0")]
```
Comment (southpolenator) [inline DebugCmdlets/Properties/AssemblyInfo.cs]: Use SharedAssemblyInfo.cs in your project to get unified version number across assemblies.

File: DebugCmdlets/DebugCmdlets.csproj [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    <DebugType>pdbonly</DebugType>
+    <Optimize>true</Optimize>
+    <OutputPath>bin\Release\</OutputPath>
+    <DefineConstants>TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="DbgEngManaged">
+      <HintPath>..\bin\Debug\DbgEngManaged.dll</HintPath>
```
Comment (southpolenator) [inline DebugCmdlets/DebugCmdlets.csproj]: Reference project, not the assembly

File: DebugCmdlets/DebugCmdlets.csproj [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    <ProjectGuid>{1E791857-FBC2-4AC2-ABD7-5CC2AA7D4D59}</ProjectGuid>
+    <OutputType>Library</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>DebugCmdlets</RootNamespace>
+    <AssemblyName>DebugCmdlets</AssemblyName>
+    <TargetFrameworkVersion>v4.5.2</TargetFrameworkVersion>
+    <FileAlignment>512</FileAlignment>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+    <DebugSymbols>true</DebugSymbols>
```
Comment (southpolenator) [inline DebugCmdlets/DebugCmdlets.csproj]: Add x64, x86 builds and also don't forget to set them in build configurations

File: DebugCmdlets/DebugCmdlets.cs [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+            {
+                Marshal.ThrowExceptionForHR(hresult);
+            }
+
+            if (SymbolPath == null)
+            {
+                SymbolPath = "srv*";
+            }
+
+            ((IDebugClient7)client).CreateProcessAndAttach(0, ProcessPath, 0x00000002);
```
Comment (southpolenator) [inline DebugCmdlets/DebugCmdlets.cs]: Set symbol path before attaching too. It helps with local symbols...

File: DebugCmdlets/Properties/AssemblyInfo.cs (hunk 3 above)
Comment (LepiKlark) [inline DebugCmdlets/Properties/AssemblyInfo.cs]: Done

File: DebugCmdlets/DebugCmdlets.cs (hunk 6 above)
Comment (LepiKlark) [inline DebugCmdlets/DebugCmdlets.cs]: Done.

File: DebugCmdlets/DebugCmdlets.csproj (hunk 4 above)
Comment (LepiKlark) [inline DebugCmdlets/DebugCmdlets.csproj]: Discussed offline. This file is auto generated. I am changing the hint path not to force debug.

File: DebugCmdlets/DebugCmdlets.csproj (hunk 5 above)
Comment (LepiKlark) [inline DebugCmdlets/DebugCmdlets.csproj]: Done.

**Pull request merged**